lib/event_log.ex

```diff
@@ -0,0 +1,103 @@
+defmodule EventLog.Consumer do
+  @callback init(Map) :: no_return()
```

Comment:
Nitpick: technically it's `map()` https://hexdocs.pm/elixir/typespecs.html#basic-types

also: `no_return` means the function is going to raise, if you don't care about the return type I think `any()` is more accurate

Reply:
Good spot. I think that `Map` will say this function should accept the atom `Map` (equivalent to `:"Elixir.Map"`)
map() is the right thing. though without using dialyzer I think anything goes